### PR TITLE
Feat: Offline backend and NFCF error handling WPB-2794

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-2794" title="WPB-2794" target="_blank">InitiativeWPB-2794</a>  V1 - Federation (on-prem)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This is a main feature branch that contains changes for the offline backend and non-fully connected graph error handling.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequest.1.png
Comment (johnxnguyen) [inline wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequest.1.png:1]: this doesn't look right

Comment (johnxnguyen) [reply wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequest.1.png:1]: Fixed it

Comment (johnxnguyen) [reply wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequest.1.png:1]: @KaterinaWire can you check that the snapshots are now correct?

Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequest.1.png:1]: I'm not sure about this snapshot either.

File: wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequestFromNotTeamMember.1.png
Comment (agisilaos) [inline wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequestFromNotTeamMember.1.png:1]: Are we sure that's the correct snapshot? Getting rid of the footer view?

**Review by agisilaos — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequestFromNotTeamMember.1.png
Comment (johnxnguyen) [reply wire-ios/Wire-iOS Tests/ReferenceImages/ProfileViewControllerTests/testForIncomingRequestFromNotTeamMember.1.png:1]: What should be in the footer?

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**